csrc/quantization/utils.cuh

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+/**
+ * Quantization utilities including:
+ *   Adjusted maximum values for qtypes.
+ *   Minimum scaling factors for qtypes.
+ */
+
+#include <cmath>
+#include <torch/types.h>
+
+#ifndef USE_ROCM
+  #include <c10/util/Float8_e4m3fn.h>
+  #define MAYBE_HOST_DEVICE C10_HOST_DEVICE
+#else
+  #include <ATen/hip/HIPContext.h>
+  #include <c10/util/Float8_e4m3fn.h>
+  #include <c10/util/Float8_e4m3fnuz.h>
+  // ROCm doesn't seem to need C10_HOST_DEVICE for static constexpr
+  #define MAYBE_HOST_DEVICE
+#endif
+
+template <typename T>
+struct quant_type_max {
+  static constexpr T val() { return std::numeric_limits<T>::max(); }
+};
+
+// Using the default max value from pytorch (240.0 0x7F) will cause accuracy
+// issues when running dynamic quantization. Here use 224.0 0x7E for rocm.
+template <>
+struct quant_type_max<c10::Float8_e4m3fnuz> {
+  static constexpr c10::Float8_e4m3fnuz val() {
+    return c10::Float8_e4m3fnuz(0x7E, c10::Float8_e4m3fnuz::from_bits());
+  }
+};
+
+template <typename T>
+MAYBE_HOST_DEVICE static constexpr T quant_type_max_v =
+    quant_type_max<T>::val();
+
+template <typename T>
+struct min_scaling_factor {
+  C10_DEVICE C10_ALWAYS_INLINE static float val() {
+    return 1.0f / (quant_type_max_v<T> * 512.0f);
```

Comment:
Sorry for being pedantic, but did you try first explicitly converting the max value
`quant_type_max_v<T>` (in fp8) to float? If the constructor is constexpr (above), I think the cast might be as well, even if the `operator*`
 isn't.

Reply:
```
/workspace/vllm/csrc/quantization/utils.cuh:44:27: note: non-constexpr function 'operator float' cannot be used in a constant expression
   44 |     return 1.0f / ((float)quant_type_max_v<T> * 512.0f);
      |                           ^
/workspace/vllm/csrc/quantization/utils.cuh:57:5: note: in call to 'val()'
   57 |     min_scaling_factor<T>::val();
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/lib/python3.12/dist-packages/torch/include/c10/util/Float8_e4m3fn-inl.h:21:39: note: declared here
   21 | inline C10_HOST_DEVICE Float8_e4m3fn::operator float() const {
```

Reply:
Okay, thanks for checking!